services/director-v2/src/simcore_service_director_v2/utils/dask.py

```diff
@@ -462,6 +462,46 @@ async def check_maximize_workers(cluster: dask_gateway.GatewayCluster | None) ->
         await cluster.scale(_LARGE_NUMBER_OF_WORKERS)
 
 
+def _can_task_run_on_worker(
+    task_resources: dict[str, Any], worker_resources: dict[str, Any]
+) -> bool:
+    def gen_check(
+        task_resources: dict[str, Any], worker_resources: dict[str, Any]
+    ) -> Iterable[bool]:
+        for name, required_value in task_resources.items():
+            if required_value is None:
+                yield True
+            elif worker_has := worker_resources.get(name):
+                yield worker_has >= required_value
+            else:
+                yield False
+
+    return all(gen_check(task_resources, worker_resources))
+
+
+def _cluster_missing_resources(
+    task_resources: dict[str, Any], cluster_resources: dict[str, Any]
+) -> list[str]:
+    return [r for r in task_resources if r not in cluster_resources]
+
+
+def _human_readable_resources(resources: dict[str, Any]) -> dict[str, Any]:
+    human_readable_resources = {}
+
+    for res_name, res_value in resources.items():
+        if "RAM" in res_name:
+            try:
+                human_readable_resources[res_name] = parse_obj_as(
+                    ByteSize, res_value
+                ).human_readable()
+            except ValidationError:
+                logger.warning("could not parse %s:%s", f"{res_name=}", res_value)
```

Comment:
MINOR: 
- Do we really need to log a warning in a function that composes info for a message? 
- rename as `_logger`

Reply:
I improved the warning message.
Yes we do as I cannot guarantee that dask does not change these messages. I will add a test.